tests/test_data.py

```diff
@@ -62,33 +223,6 @@ def test_basic_json(lexer_json):
     assert list(lexer_json.get_tokens(fragment)) == tokens
 
 
-def test_json_escape_backtracking(lexer_json):
```

Comment:
Any particular reason for removing this? The new lexer is no longer prone to this issue, as it's not using a regex, but keeping the test around doesn't hurt, and it prevents a regression. I suspect it's probably enough to check that the test terminates quickly, not the specific tokens.

Reply:
I removed it because it didn't apply. However, it won't be difficult to add back in.

I don't anticipate having time to add the pathological backtracking test back over the next few days but will as soon as I can.

Reply:
I've added the test back in, but I've asserted that it should take less than a second to run, rather than testing what the output is.

It may be worthwhile to use the pytest-timeout plugin to test catastrophic backtracking. Then, all such tests can be decorated with something like this:

```python
@pytest.mark.timeout(1)
def test_backtracking(lexer):
    list(lexer.get_tokens("input_text_with_backtracking_test"))
```

This would force all such tests to fail if they take too long, rather than waiting for someone to notice that a test is taking a long time.